cns/singletenantcontroller/reconciler.go

```diff
@@ -37,18 +38,24 @@ type Reconciler struct {
 	cnscli             cnsClient
 	ipampoolmonitorcli ipamPoolMonitorClient
 	nnccli             nncGetter
+	started            chan interface{}
+	once               sync.Once
 }
 
 func NewReconciler(nnccli nncGetter, cnscli cnsClient, ipampipampoolmonitorcli ipamPoolMonitorClient) *Reconciler {
 	return &Reconciler{
 		cnscli:             cnscli,
 		ipampoolmonitorcli: ipampipampoolmonitorcli,
 		nnccli:             nnccli,
+		started:            make(chan interface{}),
 	}
 }
 
 // Reconcile is called on CRD status changes
 func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	defer func() {
+		r.once.Do(func() { close(r.started) })
```

Comment:
Does this gets set even if we are returning error?

Reply:
moved it to the end